com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/FileSystemServiceTest.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ *
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * Broadcom, Inc. - initial API and implementation
+ *
+ */
+package com.ca.lsp.cobol.service;
+
+import com.broadcom.lsp.cdi.LangServerCtx;
+import com.broadcom.lsp.domain.cobol.event.api.EventObserver;
+import com.broadcom.lsp.domain.cobol.event.model.FetchedCopybookEvent;
+import com.broadcom.lsp.domain.cobol.event.model.RequiredCopybookEvent;
+import com.ca.lsp.cobol.FileSystemConfiguration;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Comparator;
+import java.util.List;
+
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.*;
+
+@Slf4j
+public class FileSystemServiceTest extends FileSystemConfiguration
+    implements EventObserver<RequiredCopybookEvent> {
+
+  private static final String DEP_EXTENSION = ".dep";
+  private FileSystemServiceImpl fileSystemService =
+      (FileSystemServiceImpl) LangServerCtx.getInjector().getInstance(FileSystemService.class);
```

Comment:
Do we need a class cast here? I think it will work if you call the impl class. 
Still, the best practice is to interact only with the interface methods.

Reply:
I will refactor this using mockito with a proper issue